workflowparsers/quantum_espresso_bands/parser.py

```diff
@@ -0,0 +1,570 @@
+            # case enumeration for `symmetry_window`:
+            # 1. (None, None, None) -> not possible
+            # 2. (None, None, X) -> first step (add to initial bucket)
+            # 3. (X, None, None) -> not possible
+            # 4. (X, Y, None) -> end reached
+            # 5. (None, X, Y) -> add to initial bucket
+            # 6. (X, X, Y) -> add Y to latest bucket
+            # 7. (X, Y, Y) -> add Y to latest bucket
+            # 8. (X, Y, X) -> add Y and X a new, latest bucket
+            # 9. (X, Y, Z) -> add Y and Z a new, latest bucket
+            if (None not in symmetry_window) and all(
+                [symmetry_window[i] != symmetry_window[i + 1] for i in range(2)]
+            ):
+                segments.append([previous_kpoint])
+            segments[-1].append(kpoint)
+            previous_kpoint = kpoint
+        return segments
+
+    @staticmethod
+    def apply_multiplicity(
+        energies: list[list[float]], mults: list[list[int]]
+    ) -> list[list[float]]:
+        """Apply band multiplicity to energies."""
+        return [
+            [e for e, m in zip(energy, mult) for _ in range(m)]
+            for energy, mult in zip(energies, mults)
+        ]
+
+
+class QuantumEspressoBandsParser:
+    """Parser for Quantum ESPRESSO BANDS calculations."""
+
+    def __init__(self):
+        self.bands_parser = BandsFileParser()
+        self.pwscf_parser = MainfileParser()
+
+    def _find_files(self) -> Optional[str]:
+        """Find BANDS and PWSCF files in the directory."""
+        pw_patt = re.compile(r'Program PWSCF')
+        scf_patt = re.compile(r'Self-consistent Calculation')
+
+        out_files = self.bands_parser.scan_dir_for_files(self.maindir)
+        pwscf_file = ''
+
+        for filepath in out_files:
+            with open(filepath, 'r') as f:
+                f.readline()  # Skip the first line
+                if not pw_patt.search(f.readline()):
+                    continue
+
+                for line in f:
+                    if scf_patt.search(line):
+                        if not pwscf_file:
+                            pwscf_file = filepath
+                        else:
+                            raise ValueError(
+                                f'Multiple PWSCF files with self-consistent calculations found in {self.maindir}. '
+                                f'Cannot determine which one to use for electronic structure information. '
+                                f'Files: {", ".join([pwscf_file, filepath])}'
+                            )
+
+        return pwscf_file if pwscf_file else None
+
+    def parse(self, filepath, archive, logger):
+        """
+        Parse both PWSCF output and BANDS output files.
+
+        Args:
+            filepath: Path to the main output file
+            archive: NOMAD archive where results will be stored
+            logger: Logger for output messages
+        """
+        self.bands_parser.mainfile = filepath
```

Comment:
this should be your mainile parser, since you are matching Program Bands

Reply:
How about I call the classes just by the name of the files that they parse?

Reply:
I do not quite get what you mean by this, you should not redefine the pwscf parser it already exists in electronicparsers, you should not parse any pwscf mainfile, just bands.